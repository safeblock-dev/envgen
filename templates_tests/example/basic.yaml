groups:
  - name: App
    description: Basic application settings
    fields:
      - name: debug
        type: bool
        description: Enable debug mode
        default: "false"
      - name: port
        type: int
        description: Server port
        required: true
        example: "8080"
      - name: host
        type: string
        description: Server host
        default: "localhost"

  - name: Database
    description: Database connection settings
    prefix: DB_
    fields:
      - name: url
        type: string
        description: Database connection URL
        required: true
        example: "postgres://user:pass@localhost:5432/db"
      - name: log_level
        type: LogLevel
        description: Database logging level
        default: "info"

types:
  - name: LogLevel
    description: Log level for the application
    values: [debug, info, warn, error] 