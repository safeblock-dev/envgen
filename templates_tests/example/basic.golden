# Generated by envgen. DO NOT EDIT.
# This file was automatically generated and should not be modified manually.

# --------------------------------
# App
# Basic application settings
# --------------------------------

# Enable debug mode
DEBUG=false

# Server port (required)
PORT=8080

# Server host
HOST=localhost

# --------------------------------
# Database
# Database connection settings
# --------------------------------

# Database connection URL (required)
DB_URL=postgres://user:pass@localhost:5432/db

# Database logging level [debug, info, warn, error]
DB_LOG_LEVEL=info